project/Protobuf.scala

```diff
@@ -38,7 +38,7 @@ object Protobuf {
     outputPaths := Seq((Compile / sourceDirectory).value, (Test / sourceDirectory).value).map(_ / "java"),
     importPath := None,
     // this keeps intellij happy for files that use the shaded protobuf
-    Compile / unmanagedJars += (LocalProject("protobuf-v3") / assembly).value,
+    Compile / unmanagedJars += (LocalProject("protobuf-v3") / Compile / packageBin).value,
```

Comment:
This line would be execute osgi bundle unexpected?

Reply:
This to me also looks suspicious, @Duhemm whats the context behind this change?

Reply:
`Compile / packagebin` is cached, whereas `assembly` is not. Both tasks produce the same output:

On main:
```
pekko > show protobuf-v3 / Compile / assembly
(...)
    [info] /Users/martin/Documents/Projects.nosync/apache/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/pekko-protobuf-v3-assembly-1.1.0-M0+329-231b9a99-SNAPSHOT.jar
$ md5sum < /Users/martin/Documents/Projects.nosync/apache/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/pekko-protobuf-v3-assembly-1.1.0-M0+329-231b9a99-SNAPSHOT.jar
87fdacb73c114cf95d088cb31c164622  -

pekko > show protobuf-v3 / Compile / packageBin
(...)
    [info] /Users/martin/Documents/Projects.nosync/apache/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/stripped/pekko-protobuf-v3-assembly-1.1.0-M0+329-231b9a99-SNAPSHOT.jar
$ md5sum < /Users/martin/Documents/Projects.nosync/apache/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/stripped/pekko-protobuf-v3-assembly-1.1.0-M0+329-231b9a99-SNAPSHOT.jar
87fdacb73c114cf95d088cb31c164622  -
```

With this pull request:
```
pekko > show protobuf-v3 / Compile / assembly
(...)
[info] /Users/martin/Documents/Projects.nosync/apache/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/pekko-protobuf-v3-assembly-1.1.0-M0+330-f4e48f91-SNAPSHOT.jar
$ md5sum < /Users/martin/Documents/Projects.nosync/apache/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/pekko-protobuf-v3-assembly-1.1.0-M0+330-f4e48f91-SNAPSHOT.jar
aec27c8ab1a2dd13fa5b2f15044089d1  -

pekko > show protobuf-v3 / Compile / packageBin
(...)
[info] /Users/martin/Documents/Projects.nosync/apache/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/stripped/pekko-protobuf-v3-assembly-1.1.0-M0+330-f4e48f91-SNAPSHOT.jar
$ md5sum < /Users/martin/Documents/Projects.nosync/apache/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/stripped/pekko-protobuf-v3-assembly-1.1.0-M0+330-f4e48f91-SNAPSHOT.jar
aec27c8ab1a2dd13fa5b2f15044089d1  -
```

So depending on `packageBin` rather than `assembly` should produce the same result without requiring to re-build the jar over and over again.

What's your concern with this change?

Reply:
> What's your concern with this change?

Its just that this area of pekko has historically been quite brittle largely due to how sbt + sbt-osgi is designed (see https://github.com/sbt/sbt-osgi/issues/104).

I am quite sure that this is a non issue, its just we have been burned in the past with subtle breakages but @Roiocam has added regression tests so I will approve the PR.

Reply:
protobufV3 uses a specific osgi packaging approache instead of relying on packageBin, so I have no problem with this PR.